contracts/go/randomnessgenerator/agent/main.go

```diff
@@ -277,7 +289,7 @@ func waitForTransaction(client *ethclient.Client, tx *types.Transaction) error {
 	if err != nil {
 		return err
 	}
-	return fmt.Errorf("tx failed, revert reason: %s", revertReason)
+	return fmt.Errorf("tx[%s] failed, revert reason: %s", tx.Hash().String(), revertReason)
```

Comment:
try to avoid embedding formatted strings into error messages, as it's very hard to debug after (string matching doesn't work properly). instead, consider moving the error to the end of the message, eg "tx reverted: hash: %s, reason: %s"